prow/github/client.go

```diff
@@ -319,23 +322,28 @@ func (c *Client) requestRetry(method, path, accept string, body interface{}) (*h
 						// Sleep an extra second plus how long GitHub wants us to
 						// sleep. If it's going to take too long, then break.
 						sleepTime := c.time.Until(time.Unix(int64(t), 0)) + time.Second
-						if sleepTime > 0 && sleepTime < maxSleepTime {
+						if sleepTime < maxSleepTime {
 							c.time.Sleep(sleepTime)
 						} else {
+							err = fmt.Errorf("sleep time for token reset exceeds max sleep time (%v > %v)", sleepTime, maxSleepTime)
+							resp.Body.Close()
 							break
 						}
+					} else {
+						err = fmt.Errorf("failed to parse rate limit reset unix time %q: %v", resp.Header.Get("X-RateLimit-Reset"), err)
+						resp.Body.Close()
+						break
 					}
 				} else if oauthScopes := resp.Header.Get("X-Accepted-OAuth-Scopes"); len(oauthScopes) > 0 {
 					err = fmt.Errorf("is the account using at least one of the following oauth scopes?: %s", oauthScopes)
+					resp.Body.Close()
 					break
 				}
-				resp.Body.Close()
 			} else if resp.StatusCode < 500 {
```

Comment:
Should we start handling this as elsewhere, ie. check that resp.StatusCode is 2xx?

Reply:
That is handled by the caller of this function, `requestRaw` using the `request.exitCodes` field. We can't throw an error if the status is non-2xx because github returns non-2xx statuses as a valid response for valid requests for some endpoints (see [IsMember](https://github.com/kubernetes/test-infra/blob/334afce975c037e3f1a247d44df332b70c8fd0e3/prow/github/client.go#L420-L443)).